### PR TITLE
[Merged by Bors] - Support version 377

### PR DESCRIPTION
**Pull request by maltesander**
## Description

- supporting version 377
- updated to operator-rs 0.17
- updated to config changes in version 377
- updated examples and regorules

Integration test PR: https://github.com/stackabletech/integration-tests/pull/225

closes https://github.com/stackabletech/trino-operator/issues/187

## Review Checklist
- [ ] Code contains useful comments
- [ ] (Integration-)Test cases added (or not applicable)
- [ ] Documentation added (or not applicable)
- [ ] Changelog updated (or not applicable)
- [ ] Cargo.toml only contains references to git tags (not specific commits or branches)
- [ ] Helm chart can be installed and deployed operator works (or not applicable)

Once the review is done, comment `bors r+` (or `bors merge`) to merge. [Further information](https://bors.tech/documentation/getting-started/#reviewing-pull-requests)

### CODE REVIEW

File: docs/modules/ROOT/partials/supported-versions.adoc [hunk 1]
```diff
@@ -2,4 +2,4 @@
 // This is a separate file, since it is used by both the direct Trino documentation, and the overarching
 // Stackable Platform documentation.
 
-- 362
```
Comment (sbernauer) [inline docs/modules/ROOT/partials/supported-versions.adoc:5]: Ok it seems we drop support for Trino 362.
Just want to make sure if everybody is fine with that? I personally don't have a string opinion.

File: rust/operator-binary/src/main.rs [hunk 2]
```diff
@@ -31,14 +32,19 @@ struct Opts {
 
 #[tokio::main]
 async fn main() -> anyhow::Result<()> {
-    stackable_operator::logging::initialize_logging("TRINO_OPERATOR_LOG");
+    stackable_operator::logging::initialize_logging(
+        "TRINO_OPERATOR_LOG",
+        APP_NAME,
+        TracingTarget::None,
```
Comment (sbernauer) [inline rust/operator-binary/src/main.rs:38]: Please have a look here on how to handle the tracing_target
https://github.com/stackabletech/commons-operator/blob/75418fb2bb8b58ba27db25f74f5e1020427728ee/rust/operator-binary/src/main.rs#L26-L43

Comment (maltesander) [reply rust/operator-binary/src/main.rs]: oh yeah thanks done!

File: docs/modules/ROOT/partials/supported-versions.adoc (hunk 1 above)
Comment (maltesander) [reply docs/modules/ROOT/partials/supported-versions.adoc:5]: Yes i discussed that with @lfrancke.

**Review by sbernauer — APPROVED** [review]

**Comment by maltesander** [comment]
bors merge

**Pull request closed without merging**